loading.bs

```diff
@@ -988,6 +1057,87 @@ A [=request=] |browserRequest| <dfn>matches the stored exchange</dfn>
     Behavior=] returning a list of lists.
 1. Return "match".
 
+<h3 algorithm id="create-a-new-report">Create a new signed exchange report</h3>
+To <dfn>create a new signed exchange report</dfn> with |request| and
+|actualResponse|, the UA MUST:
+
+1. Let |report| be a new [=signed exchange report=] struct.
+1. Set |report|'s [=signed exchange report/outer request=] to |request|.
+1. Set |report|'s [=signed exchange report/outer response=] to |actualResponse|.
+1. Set |report|'s [=signed exchange report/server IP=] to the IP address of the
+    server from which the user agent received the |actualResponse|, if
+    available.
+1. Return |report|.
+
+<h3 algorithm id="wait-and-queue-a-report">Wait and queue a report</h3>
+To <dfn>wait and queue a report for</dfn> |parsedExchange| and |report|, the UA
+MUST:
+
+1. Wait until |parsedExchange|'s [=response/body=]'s [=body/stream=] is
+    [=ReadableStream/closed=] or [=ReadableStream/errored=].
+1. If |parsedExchange|'s [=response/body=]'s [=body/stream=] is
+    [=ReadableStream/closed=], run [=queue a signed exchange report=] |report|
+    with "[=signed exchange report/ok=]" as the result and abort these steps.
+1. If |parsedExchange|'s [=response/body=]'s [=body/stream=] is
+    [=ReadableStream/errored=], run [=queue a signed exchange report=] |report|
+    with "[=signed exchange report/mi_error=]" as the result.
+
+<h3 algorithm id="queue-report">Queuing signed exchange report</h3>
+
+To <dfn>queue a signed exchange report</dfn> |report| and |result|, the UA MUST:
+
+1. Set |report|'s [=signed exchange report/result=] to |result|.
+
+1. Let |report body| and |policy| be the result of
+    [=generate a network error report=]</a> with |report|'s
+    [=signed exchange report/outer request=]. If the result is null, abort
+    these steps.
+
+1. If |report body|'s `"type"` is `"dns.address_changed"`, abort these steps.
+
+    Note: This means that the NEL report was downgraded because the IP addresses
+    of the server and the |policy| don't match. In this case,
+    [=deliver a network report=] algorithm MUST be called with the error report
```

Comment:
Don't use normative language like "MUST" in a Note, and try to avoid passive voice ("be called") in general. Instead, describe what happens, and ideally link to where it happens. So, what algorithm winds up calling [deliver a network report](https://github.com/w3c/network-error-logging/pull/100) in this case?